src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,61 @@
+import { nanoid } from "nanoid";
+import PropTypes from "prop-types";
+import React, { ReactElement, useRef } from "react";
+
+import "./modal.scss";
+
+type Props = {
+  buttons?: ReactElement[];
```

Comment:
Do we want this to be limited to Buttons?

Reply:
Good call, I suppose technically it can be any react element.